packages/editor/src/components/block-settings-menu/reusable-block-convert-button.js

```diff
@@ -50,31 +50,23 @@ export function ReusableBlockConvertButton( {
 export default compose( [
 	withSelect( ( select, { clientIds } ) => {
 		const {
-			getBlock,
+			getBlocksByClientId,
 			canInsertBlockType,
 			__experimentalGetReusableBlock: getReusableBlock,
 		} = select( 'core/editor' );
-		const {
-			getFreeformFallbackBlockName,
-			getUnregisteredFallbackBlockName,
-		} = select( 'core/blocks' );
 
-		const blocks = map( clientIds, ( clientId ) => getBlock( clientId ) );
+		const blocks = getBlocksByClientId( clientIds );
 
 		const isVisible = (
-			// Guard against the case where a regular block has *just* been converted to a
-			// reusable block and doesn't yet exist in the editor store.
-			every( blocks, ( block ) => !! block ) &&
-
 			// Hide 'Add to Reusable Blocks' when Reusable Blocks are disabled, i.e. when
 			// core/block is not in the allowed_block_types filter.
 			canInsertBlockType( 'core/block' ) &&
-
-			// Hide 'Add to Reusable Blocks' on Classic blocks. Showing it causes a
-			// confusing UX, because of its similarity to the 'Convert to Blocks' button.
-			( blocks.length !== 1 || (
-				blocks[ 0 ].name !== getFreeformFallbackBlockName() &&
-				blocks[ 0 ].name !== getUnregisteredFallbackBlockName()
+			// Guard against the case where a regular block has *just* been converted to a
```

Comment:
This comment now only really covers part of what's being done with the `every` condition. We might want to enhance it slightly, or even move it to cover the specific parts of the predicate for which it is relevant (`!! block && block.isValid`, with complementing comment for the `hasBlockSupport` check).